lib/src/arithmetic/floating_point/floating_point_values/floating_point_value.dart

```diff
@@ -119,135 +118,159 @@ class FloatingPointValue implements Comparable<FloatingPointValue> {
   /// representation and the size of the exponent and mantissa
-  factory FloatingPointValue.ofBigInts(BigInt exponent, BigInt mantissa,
-      {int exponentWidth = 0, int mantissaWidth = 0, bool sign = false}) {
-    final (signLv, exponentLv, mantissaLv) = (
-      LogicValue.ofBigInt(sign ? BigInt.one : BigInt.zero, 1),
-      LogicValue.ofBigInt(exponent, exponentWidth),
-      LogicValue.ofBigInt(mantissa, mantissaWidth)
-    );
-
-    return FloatingPointValue(
-        sign: signLv, exponent: exponentLv, mantissa: mantissaLv);
-  }
+  FloatingPointValue.ofBigInts(BigInt exponent, BigInt mantissa,
+      {int exponentWidth = 0, int mantissaWidth = 0, bool sign = false})
+      : this(
+            sign: LogicValue.ofBigInt(sign ? BigInt.one : BigInt.zero, 1),
+            exponent: LogicValue.ofBigInt(exponent, exponentWidth),
+            mantissa: LogicValue.ofBigInt(mantissa, mantissaWidth));
 
   /// [FloatingPointValue] constructor from a set of [int]s of the binary
   /// representation and the size of the exponent and mantissa
-  factory FloatingPointValue.ofInts(int exponent, int mantissa,
-      {int exponentWidth = 0, int mantissaWidth = 0, bool sign = false}) {
-    final (signLv, exponentLv, mantissaLv) = (
-      LogicValue.ofBigInt(sign ? BigInt.one : BigInt.zero, 1),
-      LogicValue.ofBigInt(BigInt.from(exponent), exponentWidth),
-      LogicValue.ofBigInt(BigInt.from(mantissa), mantissaWidth)
-    );
-
-    return FloatingPointValue(
-        sign: signLv, exponent: exponentLv, mantissa: mantissaLv);
-  }
-
-  /// Constructor enabling subclasses.
-  FloatingPointValue.withConstraints(
-      {required this.sign,
-      required this.exponent,
-      required this.mantissa,
-      int? mantissaWidth,
-      int? exponentWidth})
-      : value = [sign, exponent, mantissa].swizzle(),
-        _bias = computeBias(exponent.width),
-        _minExp = computeMinExponent(exponent.width),
-        _maxExp = computeMaxExponent(exponent.width) {
-    if (sign.width != 1) {
-      throw RohdHclException('FloatingPointValue: sign width must be 1');
-    }
-    if (mantissaWidth != null && mantissa.width != mantissaWidth) {
-      throw RohdHclException(
-          'FloatingPointValue: mantissa width must be $mantissaWidth');
-    }
-    if (exponentWidth != null && exponent.width != exponentWidth) {
-      throw RohdHclException(
-          'FloatingPointValue: exponent width must be $exponentWidth');
-    }
-  }
+  FloatingPointValue.ofInts(int exponent, int mantissa,
+      {int exponentWidth = 0, int mantissaWidth = 0, bool sign = false})
+      : this(
+            sign: LogicValue.ofBigInt(sign ? BigInt.one : BigInt.zero, 1),
+            exponent: LogicValue.ofBigInt(BigInt.from(exponent), exponentWidth),
+            mantissa:
+                LogicValue.ofBigInt(BigInt.from(mantissa), mantissaWidth));
 
   /// Construct a [FloatingPointValue] from a Logic word
   factory FloatingPointValue.fromLogic(
```

Comment:
Why is this called `fromLogic` instead of `fromLogicValue`?  Also, the comment says "word"? Also, this shouldn't be a factory?

Reply:
I'm struggling with having both factory and non-factory constructors.  For example, fromDouble() is a factory constructor as it does a lot of work before it can initialize fields.  But that forces sub-classes to also have a factory method that first checks for some things and then calls the base factory method, possibly then checking for other things.